File: data/python/MAXIMUM_SUBARRAY_SUM_USING_PREFIX_SUM_TRANSLATED.java
====================
public static double fGold ( int [ ] arr , int n ) {
  double minPrefixSum = 0 ;
  double res = - Double . MAX_VALUE ;
  int [ ] prefixSum = new int [ n ] ;
  prefixSum [ 0 ] = arr [ 0 ] ;
  for ( int i = 1 ;
  i < n ;
  i ++ ) {
    prefixSum [ i ] = prefixSum [ i - 1 ] + arr [ i ] ;
  }
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    res = Math . max ( res , prefixSum [ i ] - minPrefixSum ) ;
    minPrefixSum = Math . min ( minPrefixSum , prefixSum [ i ] ) ;
  }
  return res ;
}

File: data/java/MAXIMUM_SUBARRAY_SUM_USING_PREFIX_SUM_TRANSLATED.py
====================
def f_gold ( arr , n ) :
    min_prefix_sum = 0
    res = int ( '-1' )
    prefix_sum = [ arr [ 0 ] ]
    for i in range ( 1 , n ) :
        prefix_sum [ i ] = prefix_sum [ i - 1 ] + arr [ i ]
    for i in range ( 0 , n ) :
        res = max ( res , prefix_sum [ i ] - min_prefix_sum )
        min_prefix_sum = min ( min_prefix_sum , prefix_sum [ i ] )
    return res

